polyfill/index.d.ts

```diff
@@ -564,7 +564,7 @@ export namespace Temporal {
   };
 
   type DateFields = {
-    era: string | undefined;
+    era?: string | undefined;
```

Comment:
I think we should remove `era` completely from the base types. After we release to reviewers, I'm happy to take a look at how we can improve TS types for non-ISO calendars, but in the meantime it doesn't make sense to have the all-calendars TS types include one property from non-ISO calendars but not other common ones like `yearType`.

Reply:
Added another commit removing these entirely from the TypeScript types.